lib/full-site-editing/template-loader.php

```diff
@@ -69,6 +70,19 @@ function gutenberg_override_query_template( $template, $type, array $templates =
 	$current_template_slug       = basename( $template, '.php' );
 	$current_block_template_slug = is_object( $current_template ) ? $current_template->slug : false;
 	foreach ( $templates as $template_item ) {
+
+		// if the theme is a child theme we want to check if a php template exists
+		// and that a corresponding block template from the theme and not the parent doesn't exist
+		$has_php_template   = file_exists( get_stylesheet_directory() . '/' . $type . '.php' );
```

Comment:
is this a better option than using $template?

Reply:
we could use either or. $template will pull from the parent if the child doesn't have a template defined for a specific page since it uses locate_template(). With get_stylesheet_directory() we will be sure the path is not from the parent.